### PR TITLE
MAT-8702: Update HR for HTML

### PR DESCRIPTION
**Pull request by chubert-sb**
## MADiE PR

Jira Ticket: [MAT-8702](https://jira.cms.gov/browse/MAT-8702)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/resources/templates/humanreadable/measure_information_table.ftl [hunk 1]
```diff
@@ -123,7 +123,7 @@
             <td style="width:80%" colspan="3">
                 <div style="width:660px;">
                     <#if model.measureInformation.measureScoring?? && (model.measureInformation.measureScoring?lower_case != "ratio" || model.measureInformation.qdmVersion < 5.5)>
-                        <pre>${model.measureInformation.stratification!""}</pre>
+                        <pre>${(model.measureInformation.stratification?no_esc)!""}</pre>
```
Comment (adongare) [inline src/main/resources/templates/humanreadable/measure_information_table.ftl:126]: could you try once this:
```
Malicious input:
1. <script>alert(1)</script>
2. <img src="x" onerror="alert('XSS')">
3. <a href="javascript:alert('XSS')">Test</a>
4. <iframe src="javascript:alert('XSS')"></iframe>
5. <svg onload="alert('XSS')"></svg>
6. <div onclick="alert('XSS')">TEST</div>
```

Comment (adongare) [reply src/main/resources/templates/humanreadable/measure_information_table.ftl:126]: @riddhi-desai verified this. This can be exploited. we need to escape XSS tags before sending it to freemarker.

Comment (chubert-sb) [reply src/main/resources/templates/humanreadable/measure_information_table.ftl:126]: Screenshot 2025-07-07 at 9 44 06 PM
This is how it shows up in the human readable

Comment (adongare) [reply src/main/resources/templates/humanreadable/measure_information_table.ftl:126]: yea thats weird it works when entered through editor. however, when request is intercepted and input is altered, it failed. I think same will happen if you use APIs directly..didn't verify. @riddhi-desai  used some intercepter tool to manipulate the request.

Comment (riddhi-desai) [reply src/main/resources/templates/humanreadable/measure_information_table.ftl:126]: Yes, this appears server side, not through UI. So, when directly entering the payload on the PUT request, it wont be seen in the UI (The field will show empty) but when exported, human readable will execute it. It will not display the payload in the human readable

Comment (chubert-sb) [reply src/main/resources/templates/humanreadable/measure_information_table.ftl:126]: Worked with riddhi a bit and decided to add JSoup to sanitize the strings that are using the enriched text fields. This should remove anything harmful before sending to freemarker. We can add on to the policy as we find any more for our needs.

Comment (adongare) [reply src/main/resources/templates/humanreadable/measure_information_table.ftl:126]: this is nice! I was also playing with it for my task. This is the policy I came up with that covers pretty much all the tags that we are going to allow:

```
Safelist.basic()
              .addTags(
                  "p",
                  "strong",
                  "em",
                  "u",
                  "s",
                  "ul",
                  "ol",
                  "li",
                  "br",
                  "table",
                  "tbody",
                  "td",
                  "tfoot",
                  "th",
                  "thead",
                  "tr",
                  "col",
                  "colgroup")
              .addAttributes("table", "style")
              .addAttributes("th", "rowspan", "colspan", "style", "colwidth")
              .addAttributes("td", "rowspan", "colspan", "style", "colwidth")
              .addAttributes("col", "style");
```

**Review by adongare — DISMISSED** [review]

**Review by adongare — APPROVED** [review]
